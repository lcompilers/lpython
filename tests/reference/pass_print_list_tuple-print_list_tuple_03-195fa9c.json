{
    "basename": "pass_print_list_tuple-print_list_tuple_03-195fa9c",
    "cmd": "lpython --pass=print_list_tuple --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/print_list_tuple_03.py",
    "infile_hash": "5131d7b6049e22c278ad41494969ff1f70c1dc811313032a9fcf58f3",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "pass_print_list_tuple-print_list_tuple_03-195fa9c.stdout",
    "stdout_hash": "edb9d31c77ac27a72de4454275693936ef43c07263a2da687e16da5c",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}