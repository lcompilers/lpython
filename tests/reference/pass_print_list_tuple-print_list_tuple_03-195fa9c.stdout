(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            __main__global_stmts:
                                (Function
                                    (SymbolTable
                                        4
                                        {
                                            
                                        })
                                    __main__global_stmts
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [f]
                                    []
                                    [(SubroutineCall
                                        2 f
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            f:
                                (Function
                                    (SymbolTable
                                        3
                                        {
                                            __list_iterator:
                                                (Variable
                                                    3
                                                    __list_iterator
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            __list_var:
                                                (Variable
                                                    3
                                                    __list_var
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (List
                                                        (Integer 4)
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            x:
                                                (Variable
                                                    3
                                                    x
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Dict
                                                        (Integer 4)
                                                        (Tuple
                                                            [(Integer 4)
                                                            (Integer 4)]
                                                        )
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            y:
                                                (Variable
                                                    3
                                                    y
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Dict
                                                        (Integer 4)
                                                        (List
                                                            (Integer 4)
                                                        )
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    f
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(Assignment
                                        (Var 3 x)
                                        (DictConstant
                                            []
                                            []
                                            (Dict
                                                (Integer 4)
                                                (Tuple
                                                    [(Integer 4)
                                                    (Integer 4)]
                                                )
                                            )
                                        )
                                        ()
                                    )
                                    (Assignment
                                        (Var 3 x)
                                        (DictConstant
                                            [(IntegerConstant 1 (Integer 4))
                                            (IntegerConstant 2 (Integer 4))]
                                            [(TupleConstant
                                                [(IntegerConstant 1 (Integer 4))
                                                (IntegerConstant 2 (Integer 4))]
                                                (Tuple
                                                    [(Integer 4)
                                                    (Integer 4)]
                                                )
                                            )
                                            (TupleConstant
                                                [(IntegerConstant 3 (Integer 4))
                                                (IntegerConstant 4 (Integer 4))]
                                                (Tuple
                                                    [(Integer 4)
                                                    (Integer 4)]
                                                )
                                            )]
                                            (Dict
                                                (Integer 4)
                                                (Tuple
                                                    [(Integer 4)
                                                    (Integer 4)]
                                                )
                                            )
                                        )
                                        ()
                                    )
                                    (Assignment
                                        (Var 3 y)
                                        (DictConstant
                                            []
                                            []
                                            (Dict
                                                (Integer 4)
                                                (List
                                                    (Integer 4)
                                                )
                                            )
                                        )
                                        ()
                                    )
                                    (Assignment
                                        (Var 3 y)
                                        (DictConstant
                                            [(IntegerConstant 1 (Integer 4))
                                            (IntegerConstant 2 (Integer 4))]
                                            [(ListConstant
                                                [(IntegerConstant 1 (Integer 4))
                                                (IntegerConstant 2 (Integer 4))]
                                                (List
                                                    (Integer 4)
                                                )
                                            )
                                            (ListConstant
                                                [(IntegerConstant 3 (Integer 4))
                                                (IntegerConstant 4 (Integer 4))]
                                                (List
                                                    (Integer 4)
                                                )
                                            )]
                                            (Dict
                                                (Integer 4)
                                                (List
                                                    (Integer 4)
                                                )
                                            )
                                        )
                                        ()
                                    )
                                    (Print
                                        [(StringConstant
                                            "("
                                            (Character 1 1 ())
                                        )]
                                        ()
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                    )
                                    (Print
                                        [(TupleItem
                                            (DictItem
                                                (Var 3 x)
                                                (IntegerConstant 1 (Integer 4))
                                                ()
                                                (Tuple
                                                    [(Integer 4)
                                                    (Integer 4)]
                                                )
                                                ()
                                            )
                                            (IntegerConstant 0 (Integer 4))
                                            (Integer 4)
                                            ()
                                        )]
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                    )
                                    (Print
                                        [(StringConstant
                                            ", "
                                            (Character 1 2 ())
                                        )]
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                    )
                                    (Print
                                        [(TupleItem
                                            (DictItem
                                                (Var 3 x)
                                                (IntegerConstant 1 (Integer 4))
                                                ()
                                                (Tuple
                                                    [(Integer 4)
                                                    (Integer 4)]
                                                )
                                                ()
                                            )
                                            (IntegerConstant 1 (Integer 4))
                                            (Integer 4)
                                            ()
                                        )]
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                    )
                                    (Print
                                        [(StringConstant
                                            ")"
                                            (Character 1 1 ())
                                        )]
                                        ()
                                        (StringConstant
                                            " "
                                            (Character 1 1 ())
                                        )
                                    )
                                    (Print
                                        [(StringConstant
                                            "("
                                            (Character 1 1 ())
                                        )]
                                        ()
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                    )
                                    (Print
                                        [(TupleItem
                                            (DictItem
                                                (Var 3 x)
                                                (IntegerConstant 2 (Integer 4))
                                                ()
                                                (Tuple
                                                    [(Integer 4)
                                                    (Integer 4)]
                                                )
                                                ()
                                            )
                                            (IntegerConstant 0 (Integer 4))
                                            (Integer 4)
                                            ()
                                        )]
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                    )
                                    (Print
                                        [(StringConstant
                                            ", "
                                            (Character 1 2 ())
                                        )]
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                    )
                                    (Print
                                        [(TupleItem
                                            (DictItem
                                                (Var 3 x)
                                                (IntegerConstant 2 (Integer 4))
                                                ()
                                                (Tuple
                                                    [(Integer 4)
                                                    (Integer 4)]
                                                )
                                                ()
                                            )
                                            (IntegerConstant 1 (Integer 4))
                                            (Integer 4)
                                            ()
                                        )]
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                    )
                                    (Print
                                        [(StringConstant
                                            ")"
                                            (Character 1 1 ())
                                        )]
                                        ()
                                        (StringConstant
                                            " "
                                            (Character 1 1 ())
                                        )
                                    )
                                    (Assignment
                                        (Var 3 __list_var)
                                        (DictItem
                                            (Var 3 y)
                                            (IntegerConstant 1 (Integer 4))
                                            ()
                                            (List
                                                (Integer 4)
                                            )
                                            ()
                                        )
                                        ()
                                    )
                                    (Print
                                        [(StringConstant
                                            "["
                                            (Character 1 1 ())
                                        )]
                                        ()
                                        (StringConstant
                                            ""
                                            (Character 1 0 ())
                                        )
                                    )
                                    (DoLoop
                                        ()
                                        ((Var 3 __list_iterator)
                                        (IntegerConstant 0 (Integer 4))
                                        (IntegerBinOp
                                            (ListLen
                                                (Var 3 __list_var)
                                                (Integer 4)
                                                ()
                                            )
                                            Sub
                                            (IntegerConstant 1 (Integer 4))
                                            (Integer 4)
                                            ()
                                        )
                                        (IntegerConstant 1 (Integer 4)))
                                        [(Print
                                            [(ListItem
                                                (Var 3 __list_var)
                                                (Var 3 __list_iterator)
                                                (Integer 4)
                                                ()
                                            )]
                                            (StringConstant
                                                ""
                                                (Character 1 0 ())
                                            )
                                            (StringConstant
                                                ""
                                                (Character 1 0 ())
                                            )
                                        )
                                        (If
                                            (IntegerCompare
                                                (Var 3 __list_iterator)
                                                Lt
                                                (IntegerBinOp
                                                    (ListLen
                                                        (Var 3 __list_var)
                                                        (Integer 4)
                                                        ()
                                                    )
                                                    Sub
                                                    (IntegerConstant 1 (Integer 4))
                                                    (Integer 4)
                                                    ()
                                                )
                                                (Logical 4)
                                                ()
                                            )
                                            [(Print
                                                [(StringConstant
                                                    ", "
                                                    (Character 1 2 ())
                                                )]
                                                (StringConstant
                                                    ""
                                                    (Character 1 0 ())
                                                )
                                                (StringConstant
                                                    ""
                                                    (Character 1 0 ())
                                                )
                                            )]
                                            []
                                        )]
                                        []
                                    )
                                    (Print
                                        [(StringConstant
                                            "]"
                                            (Character 1 1 ())
                                        )]
                                        ()
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    __main__
                    []
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        5
                        {
                            __main__global_stmts:
                                (ExternalSymbol
                                    5
                                    __main__global_stmts
                                    2 __main__global_stmts
                                    __main__
                                    []
                                    __main__global_stmts
                                    Public
                                )
                        })
                    main_program
                    [__main__]
                    [(SubroutineCall
                        5 __main__global_stmts
                        2 __main__global_stmts
                        []
                        ()
                    )]
                )
        })
    []
)
